### PR TITLE
Th2 4649 add unit tests for etc petc

### PR DESCRIPTION
**Pull request by xHacka**

### CODE REVIEW

**Comment by xHacka** [comment]
2 Tests fail, but because of tests in `TH2-4124` branch.

**Comment by xHacka** [comment]
Main changes were made in here:
1. [conftest.py](https://github.com/th2-net/th2-data-services/blob/TH2-4649-add-unit-tests-for-etc-petc/tests/tests_unit/conftest.py)
2. [test_events_tree.py](https://github.com/th2-net/th2-data-services/blob/TH2-4649-add-unit-tests-for-etc-petc/tests/tests_unit/test_events_tree.py)
3. [test_events_tree_collection.py](https://github.com/th2-net/th2-data-services/blob/TH2-4649-add-unit-tests-for-etc-petc/tests/tests_unit/test_events_tree_collection.py)
4. [test_events_tree_parent_collection.py](https://github.com/th2-net/th2-data-services/blob/TH2-4649-add-unit-tests-for-etc-petc/tests/tests_unit/test_events_tree_parent_collection.py)

other changes came from [TH2-4674](https://exactpro.atlassian.net/browse/TH2-4674)

File: tests/tests_unit/test_events_tree_collection.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    etc = demo_etc_with_data
+    etc.append_event(
+        event={
+            "eventId": "a20f5ef4-c3fe-bb10-a29c-dd3d784909eb",
+            "parentEventId": "8e2524fa-cf59-11eb-a3f7-094f904c3a62",
+            "eventName": "StubEvent",
+        }
+    )
+    detached_events = etc.get_detached_events()
+    assert [event for event in detached_events if event["eventName"] == "StubEvent"]
```
Comment (ConnectDIY) [inline tests/tests_unit/test_events_tree_collection.py:112]: Can you check the whole tree here, too, please

File: tests/tests_unit/test_events_tree_collection.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+
+def test_etc_append_non_stub_event(demo_etc_with_data):
```
Comment (ConnectDIY) [inline tests/tests_unit/test_events_tree_collection.py:115]: I don't see the difference with `test_etc_append_stub_event`

**Review by ConnectDIY — CHANGES_REQUESTED** [review]

File: tests/tests_unit/test_events_tree_collection.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+DEMO_CHILD_ID = "88a3ee80-d1b4-11eb-b0fb-199708acc7bc"
+DEMO_PARENT_ID = "84db48fc-d1b4-11eb-b0fb-199708acc7bc"
+
+
+def test_etc_for_detached_events_with_data(demo_etc_with_data):
+    assert demo_etc_with_data.get_detached_events()  # Detached Events Exist Without Data Source
+
+
+def test_etc_for_detached_events_without_data(demo_etc):
```
[the remark this answers is not in the document]
Comment (xHacka) [reply tests/tests_unit/test_events_tree_collection.py:40]: Its demonstrated in the `get_started_example` so I decided to add a check.

File: tests/tests_unit/test_events_tree_collection.py (hunk 2 above)
Comment (xHacka) [reply tests/tests_unit/test_events_tree_collection.py:115]: Difference is `parentEventId`, Events with yet unknown parent ids are stored in detached, while known parent ids can be found in ETC

File: tests/tests_unit/test_events_tree_collection.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ConnectDIY) [reply tests/tests_unit/test_events_tree_collection.py:40]: image
I see you. To be honest, it was not empty.
In this way, let's remove it

File: tests/tests_unit/test_events_tree_collection.py (hunk 2 above)
Comment (ConnectDIY) [reply tests/tests_unit/test_events_tree_collection.py:115]: ok, see you. Let's leave it

**Review by ConnectDIY — APPROVED** [review]

**Pull request merged**